src/chess_widget.rs

```diff
@@ -165,8 +166,12 @@ impl ChessWidget {
 }
 
 impl Widget<String> for ChessWidget {
-    fn event(&mut self, ctx: &mut EventCtx, event: &Event, _data: &mut String, _env: &Env) {
+    fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut String, env: &Env) {
         match event {
+            //Let this widget receive keyboard Events
+            Event::WindowConnected => {
```

Comment:
Oh nice, so this makes keyboard events go to this program? Does this event only happen once?

Reply:
yeah, so the event is sent when the widget is added to the widget tree, so essentially when the window is created.